akka-docs/rst/java/code/jdocs/akka/typed/IntroSpec.java

```diff
@@ -0,0 +1,229 @@
+            GetSession getSession = (GetSession) msg;
+            ActorRef<PostMessage> wrapper = ctx.createAdapter(p ->
+              new PostSessionMessage(getSession.screenName, p.message));
+            getSession.replyTo.tell(new SessionGranted(wrapper));
+            sessions.add(getSession.replyTo);
+            return Actor.same();
+          } else if (msg instanceof PostSessionMessage) {
+            PostSessionMessage post = (PostSessionMessage) msg;
+            MessagePosted mp = new MessagePosted(post.screenName, post.message);
+            // Local variable postCount defined in an enclosing scope must be final or effectively final
+            //postCount += 1;
+            sessions.forEach(s -> s.tell(mp));
+            return Actor.same();
+          } else {
+            return Actor.unhandled();
+          }
+        });
+      });
+    }
+
+    public static Behavior<Command> mutableChatRoom2() {
+      // TODO Type safety: The expression of type IntroSpec.ChatRoom.MutableChatRoom2 needs unchecked conversion to
+      //      conform to Behavior<IntroSpec.ChatRoom.Command>
+      return Actor.deferred(ctx -> new MutableChatRoom2());
+    }
+
+    public static class MutableChatRoom2 extends ExtensibleBehavior<Command> {
+      final List<ActorRef<SessionEvent>> sessions = new ArrayList<ActorRef<SessionEvent>>();
+      int postCount = 0;
+
+      @Override
+      public Behavior<Command> message(akka.typed.ActorContext<Command> ctx, Command msg) {
+        if (msg instanceof GetSession) {
+          GetSession getSession = (GetSession) msg;
+          ActorRef<PostMessage> wrapper = ctx.createAdapter(p ->
+            new PostSessionMessage(getSession.screenName, p.message));
+          getSession.replyTo.tell(new SessionGranted(wrapper));
+          sessions.add(getSession.replyTo);
+          return Actor.same();
+        } else if (msg instanceof PostSessionMessage) {
+          PostSessionMessage post = (PostSessionMessage) msg;
+          MessagePosted mp = new MessagePosted(post.screenName, post.message);
+          postCount += 1;
+          sessions.forEach(s -> s.tell(mp));
+          return Actor.same();
+        } else {
+          return Actor.unhandled();
+        }
+      }
+
+      @Override
+      public Behavior<Command> management(akka.typed.ActorContext<Command> ctx, Signal msg) {
+        return Actor.same();
+      }
+    }
+
+    public static Behavior<Command> mutableChatRoom3() {
+      return Actor.mutable(ctx -> new MutableChatRoom3(ctx));
+    }
+
+    public static class MutableChatRoom3 extends Actor.MutableBehavior<Command> {
+      final ActorContext<Command> ctx;
+      final List<ActorRef<SessionEvent>> sessions = new ArrayList<ActorRef<SessionEvent>>();
+
+      public MutableChatRoom3(ActorContext<Command> ctx) {
+        this.ctx = ctx;
+      }
+
+      @Override
+      public Behavior<Command> onMessage(Command msg) {
+        if (msg instanceof GetSession) {
```

Comment:
ignore these ugly `instanceof` for now, we should provide same thing as we have for `AbstractActor`